code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/external/ExternalServiceTokenUtil.java

```diff
@@ -108,7 +119,10 @@ public Token getUserIdentityFromJWT() {
         if (jsonData == null) {
             return null;
         }
-        Object idObject = (Object)jsonData.get(USER_IDENTITY);
+        AuthToken authToken= authTokenDao.getTokenByKey(getJWT());
```

Comment:
Why do you need the methods jsonToIdentity() and getUserIdentityFromJWT() here? I believe the parent class AbstractTokenUtil already has these methods